cmd/kube-scheduler/app/options/options.go

```diff
@@ -237,15 +238,19 @@ func (o *Options) Config() (*schedulerappconfig.Config, error) {
 	}
 
 	// Prepare event clients.
-	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: eventClient.EventsV1beta1().Events("")})
-	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, c.ComponentConfig.SchedulerName)
-	leaderElectionBroadcaster := record.NewBroadcaster()
-	leaderElectionRecorder := leaderElectionBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: c.ComponentConfig.SchedulerName})
+	if _, err := client.Discovery().ServerResourcesForGroupVersion(eventsv1beta1.SchemeGroupVersion.String()); err == nil {
```

Comment:
were we making other API calls to the server in `Config()` previously?

Reply:
no we were not IIRC. If there's an error the v1beta1 related fields would be nil -> fallback. Are you concerned about adding an API call for building sheduler's config ?

Reply:
@liggitt - can you explain why do you think it's a problem?

Reply:
It removes our ability to construct Config without a live server present. It's normal to make config, then start the process, and expect API calls to not start happening or being required until Run is called

Reply:
@yastij - I see that you already moved that check to server.go - please remove it from here now.